test_package/CMakeLists.txt

```diff
@@ -0,0 +1,270 @@
+#  Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Comment on the overall test_package dir.

Maybe it is an idea to add this dir as a sub dir of examples and rename it to conan_usage_example and a small readme how this dir can be used standalone to create a project using Celix.

I am also thinking about promoting this more in the introduction to Celix documentation, but this is for a different pull request. But this could be very nice for users so that they can quickly start using Celix.

Reply:
I moved it to examples/conan_test_package, and added minimum documentation. I prefer the `test_package` in folder name because it is a pretty standard conan terminology. As explained by documentation, it serves a very specific purpose.